File: python/autoopmodes/do_nothing_auto.py
# Copyright (c) FIRST and other WPILib contributors.
# Open Source Software; you can modify and/or share it under the terms of
# the WPILib BSD license file in the root directory of this project.

from wpilib.opmode import PeriodicOpMode, autonomous


@autonomous
class DoNothingAuto(PeriodicOpMode):
    """Autonomous mode that does nothing."""

    def __init__(self, robot):
        super().__init__()
        self._robot = robot

    def disabledPeriodic(self):
        self._robot.robotPeriodic()

    def periodic(self):
        self._robot.robotPeriodic()

File: cpp/src/main/cpp/robot/autoopmodes/DoNothingAuto.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "DoNothingAuto.h"

using namespace wpi::robot::autoopmodes;

DoNothingAuto::DoNothingAuto(Robot& robot) : m_robot{robot} {}

void DoNothingAuto::DisabledPeriodic() {
  m_robot.RobotPeriodic();
}

void DoNothingAuto::Periodic() {
  m_robot.RobotPeriodic();
}
